src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/LetterAddressHelper.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.helper;
+
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.LINE_SEPARATOR;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.CommonFunction.nullToEmpty;
+
+@Component
+public class LetterAddressHelper {
+
+    private List<String> addressLines = new ArrayList<>();
+
+    public String formatAddressForLetterPrinting(Map<String, Object> address) {
```

Comment:
Is this still being used even with the template using the case data?

Reply:
Yes we still use it to build the addressee part of the case data for the documents:
"                 .formattedAddress(letterAddressHelper.formatAddressForLetterPrinting(addressToSendTo))"

Reply:
Got it. Ideally, we'd only modify the data we send to Docmosis.
Could we not make a copy of the original case data, modify it, and then set it back into the CaseDetails object?